neqo-transport/src/recv_stream.rs

```diff
@@ -63,6 +63,10 @@ impl RecvStreams {
         now: Instant,
         rtt: Duration,
     ) {
+        #[expect(
+            clippy::iter_over_hash_type,
+            reason = "OK to loop over streams in an undefined order."
```

Comment:
Our prioritization scheme depends on in-order progress in the order of creation.  It's not fatal, but we've always assumed that this is in-order.  Otherwise, you get bytes in an indeterminate order.  In particular, we don't tend to adjust priority of control streams in HTTP.

Reply:
OK, then we need to add a test for that.